SYCL/BFloat16/bfloat16_builtins.cpp

```diff
@@ -0,0 +1,251 @@
+// REQUIRES: cuda
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out -Xsycl-target-backend --cuda-gpu-arch=sm_80
+// RUN: %t.out
+
+#include <CL/sycl.hpp>
+
+#include <cmath>
+#include <vector>
+
+using namespace cl::sycl;
+using sycl::ext::oneapi::experimental::bfloat16;
+
+constexpr int N = 60; // divisible by all tested array sizes
+constexpr float bf16_eps = 0.00390625;
+
+float make_fp32(uint16_t x) {
```

Comment:
Why do we still need this function in the bfloat16 case? this should use to_float function.

Reply:
Thanks. I've switched to the implicit cast to float.

Reply:
`bfloat16.to_float(uint16_t raw)` as it is currently defined takes a raw storage type, and is a member function of bfloat16 rather than a free function: if there were to be an explicit cast I think there should be a free function added that can take bfloat16 rather than uint16_t.